test/test_syscalls.c

```diff
@@ -1957,6 +1957,90 @@ static int do_test_create_ro_dir(int flags, const char *flags_str)
 	return 0;
 }
 
+/* 	this tests open with O_TMPFILE
+	note that this will only work with the fuse low level api 
+	you will get ENOTSUP with the high level api */
+static int test_create_tmpfile(void) 
+{
+	rmdir(testdir);
+	int res = mkdir(testdir, 0777);
+	if (res)
+		return -1;
+	
+	start_test("create tmpfile");
+
+	int fd = open(testdir, O_TMPFILE | O_RDWR, S_IRUSR | S_IWUSR);
+	if(fd == -1) {
+		PERROR("open");
+		return -1;
+	}
+	close(fd);
+
+	fd = open(testdir, O_TMPFILE | O_WRONLY | O_EXCL, S_IRUSR | S_IWUSR);
+	if(fd == -1){
+		PERROR("open");
+		return -1;
+	};
+	close(fd);
+
+	fd = open(testdir, O_TMPFILE | O_RDONLY, S_IRUSR);
+	if (fd != -1) {
+		ERROR("open O_TMPFILE | O_RDONLY");
+		return -1;
+	}
+	
+	success();
+	return 0;	
+}
+
+static int test_create_and_link_tmpfile(void) 
+{
+	rmdir(testdir);
+	unlink(testfile);
+
+	int res = mkdir(testdir, 0777);
+	if (res)
+		return -1;
+
+	start_test("create and link tmpfile");
+
+	int fd = open(testdir, O_TMPFILE | O_RDWR | O_EXCL, S_IRUSR | S_IWUSR);
+	if(fd == -1) {
+		PERROR("open");
```

Comment:
Could you add a bit more descriptive error message, like `open(, O_TMPFILE`)?

Reply:
done